task-generator/remote/main.go

```diff
@@ -18,6 +18,7 @@ import (
 	"flag"
```

Comment:
I'm having a hard time trying to understand the task-generator changes. Could you explain them in the commit message?

Reply:
Sure, I can add some comments into the code.